roles/DCOS.agent_windows/tasks/main.yml

```diff
@@ -0,0 +1,121 @@
+- name: Windows | Defining DC/OS by using 'version' or, more specifically, 'image_commit'
+  set_fact:
+     dcos_version_specifier: "{{ dcos_version | default('1.13') }}"
+     ansible_password: '{{ hostvars[inventory_hostname].pass }}'
+
+
+- debug:
+    msg: "dcos_version_specifier = {{ dcos_version_specifier }} , pass = {{ ansible_password }}, bucket_name = {{ bucket_name }}"
+
+
+#- name: Windows | edit Powershell Start script with local IP address of Mesos Master
+#  replace:#
+#    path: /files/start.ps1
+#    regexp: 'local_ip_master'
+#    replace: '{{  hostvars[groups.masters[0]].inventory_hostname }}'
+#  delegate_to: localhost
+
+
+- name: Windows | create folder for temporaly files
+  file:
+    path: /files
+    state: directory
+  delegate_to: localhost
+
+
+- name: Windows | create sub folder with dcos_version
+  file:
+    path: /files/{{dcos_version_specifier}}
+    state: directory
+  delegate_to: localhost
+
+
+- name: Windows | get vc_redist.x64.exe file from s3 to localhost
+  get_url:
+    url: "http://{{ bucket_name }}/vc_redist.x64.exe"
+    dest: /files/vc_redist.x64.exe
+    mode: '0660'
+  delegate_to: localhost
+
+
+- name: Windows | get vmesos-binaries.zip from s3 to localhost
+  get_url:
+    url: "http://{{ bucket_name }}/{{ dcos_version_specifier }}/mesos-binaries.zip"
+    dest: "/files/{{ dcos_version_specifier }}/mesos-binaries.zip"
+    mode: '0440'
+  delegate_to: localhost
+
+
+- name: Windows | initialisation windows-node (with install.ps1)
+  win_copy: src=/files/{{ item }} dest=C:\
```

Comment:
why root filesystem? is there a technical reason? better stay in e.g. `/dcos` and have e.g. links in `/dcos/bin/mesos-agent -> /dcos/dist/{{ dcos_version_specifier }}/mesos-agent`

or something like this... I'm not a windows expert but when you try to do an update you will need to overwrite the files which might also lead to a complicated rollback scenario

Reply:
good point, I'm not sure about those links, but having a root folder like 

> c:\dcos

 really makes sense

Reply:
agreed. it makes sense